src/org/usfirst/frc4904/robot/subsystems/Hopper.java

```diff
@@ -0,0 +1,46 @@
+package org.usfirst.frc4904.robot.subsystems;
+
+
+import org.usfirst.frc4904.standard.commands.Idle;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+public class Hopper extends Subsystem {
+	protected final DoubleSolenoid leftSolenoid;
+	protected final DoubleSolenoid rightSolenoid;
+
+	public static enum HopperState {
+		BALLIO(DoubleSolenoid.Value.kReverse), SHOOTER(DoubleSolenoid.Value.kForward);
+		private final DoubleSolenoid.Value value;
+
+		private HopperState(DoubleSolenoid.Value value) {
+			this.value = value;
+		}
+
+		public DoubleSolenoid.Value getValue() {
+			return value;
+		}
+	}
+	protected HopperState currentState;
+
+	public HopperState getState() {
+		return currentState;
+	}
+
+	public void setHopperState(HopperState desiredState) {
```

Comment:
Nit: can we call this set or setState? Saying `hopper.setHopperState(Hopper.HopperState.SHOOTER)` seems pretty verbose.

Precedents:
- https://github.com/RoboticsTeam4904/2016-Code/blob/4a72935d41f3a91088706fe303eb7a731e3f63b4/src/org/usfirst/frc4904/robot/subsystems/RockNRoller.java#L30
- https://github.com/RoboticsTeam4904/2016-Code/blob/4a72935d41f3a91088706fe303eb7a731e3f63b4/src/org/usfirst/frc4904/robot/subsystems/RockNRoller.java#L30